es/es-repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureRepository.java

```diff
@@ -76,7 +81,9 @@
     }
 
     public static List<Setting<?>> optionalSettings() {
-        return List.of(Repository.CONTAINER_SETTING,
+        return List.of(Repository.ACCOUNT_SETTING,
```

Comment:
I've tried to support usage of all client settings as `create repository` params, but it is a bit confusing when it comes to merging of previous and current settings, some of them has default values, some not, etc. Also, for some hierarchical settings like `proxy.host` we need to adjust `RepositorySettingsModule` to be able to handle them.

It should be ok to go with support of `account` and `key` settings for now and add support for the rest when we have time. WDYT?

Reply:
Would it help to already drop the support of having the setting in the `crate.yml` ?

From a user perspective it is rather confusing that some settings work and some not.

Reply:
crate.yml support for all client specific settings? if yes, it would be nice and will simplify some things.

Reply:
yep, drop support for reading any azure repository related settings from `crate.yml`

Reply:
i've dropped support for azure repository settings form crate.yaml and add support for the same setting to the `create repository` statement params. It is a bit hard to split into two logical changes.